appinsights/datacontracts.go

```diff
@@ -138,19 +156,29 @@ func (item *MetricTelemetry) baseData() Domain {
 	return item.data
 }
 
+func (item *MetricTelemetry) SetProperty(key, value string) {
+	if item.data.Properties == nil {
+		item.data.Properties = make(map[string]string)
+	}
+	item.data.Properties[key] = value
+}
+
 type RequestTelemetry struct {
 	BaseTelemetry
 	data *requestData
 }
 
-func NewRequestTelemetry(name string, timestamp time.Time, duration time.Duration, responseCode string, success bool) *RequestTelemetry {
+func NewRequestTelemetry(name, httpMethod, url string, timestamp time.Time, duration time.Duration, responseCode string, success bool) *RequestTelemetry {
 	now := time.Now()
```

Comment:
> I think we'll still want to use RequestTelemetry 

Agree here. I'm questioning that the inside of this method seems to be constructing an envelope. Sounds like the job for a different layer. That would allow passing of objects to track methods that merely conform to an interface, since the conversion is not held within the object